### PR TITLE
Improve the UI of the add to collection widget

### LINKED ISSUE #2245: Work form, relationships tab, collections section
1. Update the Collection section to enable adding work to a collection via a select menu and button. Text label to the left of the select menu is “Add to collection”. Button is to the right of the select menu and has label “Add”.
1. Add a table below the Collection section showing what collections the work is currently part of (so the user doesn't try to add the work to a collection it is already part of). If this is a new work or an existing work that hasn't been added to any collections, omit the "This work is currently in these collections" heading and the table beneath it.

relationships-tab-changes

Ref #2034

### LINKED ISSUE #2247: Work form, relationships tab, child works section
1. Update the child work section to enable adding work to a work via a select menu and button. Text label to the left of the select menu is “Add to work”. Button is to the right of the select menu and has label “Add”.
1. Add a table below the child work section showing what works the work is currently part of (so the user doesn't try to add the work to a collection it is already part of). If this is a new work or an existing work that hasn't been added to any works, omit the "This work is currently in these works" heading and the table beneath it.

This was intended to mirror #2245 

relationships-tab-changes

Ref #2034

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #2245
Fixes #2247

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/assets/javascripts/hyrax/autocomplete.es6 [hunk 1]
```diff
@@ -18,6 +18,13 @@ export default class Autocomplete {
           element.data('exclude-work')
         )
         break
+      case 'collection':
+        new Work(
```
Comment (mjgiarlo) [inline app/assets/javascripts/hyrax/autocomplete.es6]: Renamed to `Resource`? Need test coverage for this.

Comment (mjgiarlo) [reply app/assets/javascripts/hyrax/autocomplete.es6]: Also on line 15 above.

File: app/forms/hyrax/forms/work_form.rb [hunk 2]
```diff
@@ -10,13 +10,17 @@ class WorkForm
       # This is required so that fields_for will draw a nested form.
       # See ActionView::Helpers#nested_attributes_association?
       #   https://github.com/rails/rails/blob/v5.0.2/actionview/lib/action_view/helpers/form_helper.rb#L1890
-      delegate :work_members_attributes=, to: :model
+      delegate :work_members_attributes=, :member_of_collections_attributes=, to: :model
+
+      # TODO: Remove?
```
Comment (mjgiarlo) [inline app/forms/hyrax/forms/work_form.rb]: Justin says to check out this TODO.

File: app/assets/javascripts/hyrax/autocomplete/default.es6 [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// This script inializes a jquery-ui autocomplete widget
```
Comment (mjgiarlo) [inline app/assets/javascripts/hyrax/autocomplete/default.es6]: Typo

**Comment by elrayle** [comment]
@jcoyne I don't see any tests for _form_member_of_collections.html.erb

**Pull request merged**